cloudsupport/v1/ekssupport.go

```diff
@@ -137,23 +136,25 @@ func (eksSupport *EKSSupport) GetContextName(cluster string) string {
 	return ""
 }
 
-// Getcfgmap returns the authdata containing mappings of iam-roles/groups or iam-users/groups
-func (EKSSupport *EKSSupport) GetAwsCfgMap(kapi *k8sinterface.KubernetesApi, namespace string) error {
+// GetEKSCfgMap returns the ConfigMap containing mappings of iam-roles/groups or iam-users/groups
+func (EKSSupport *EKSSupport) GetEKSCfgMap(kapi *k8sinterface.KubernetesApi, namespace string) (*v1.ConfigMap, error) {
 
-	resp, err := kapi.KubernetesClient.CoreV1().ConfigMaps(namespace).Get(context.TODO(), awsauthconfigmap, metav1.GetOptions{})
+	var authData awsAuth
+
+	eksCfgMap, err := kapi.KubernetesClient.CoreV1().ConfigMaps(namespace).Get(context.TODO(), awsauthconfigmap, metav1.GetOptions{})
 
 	if err != nil {
-		return fmt.Errorf("aws-auth config map not found")
+		return nil, err
 	}
 
-	if err := json.Unmarshal([]byte(resp.Data["mapRoles"]), &ListmapRoles); err != nil {
-		return err
+	if err := json.Unmarshal([]byte(eksCfgMap.Data["mapRoles"]), &authData.MapRoles); err != nil {
```

Comment:
You should always validate the key is in the map before accessing it ->
```suggestion
if mapRoles, ok := eksCfgMap.Data["mapRoles"]; ok {
	if err := json.Unmarshal([]byte(mapRoles), &authData.MapRoles); err != nil {
			return nil, err
		}
	}		
}
```
You should apply a similar change on the `mapUsers` as well

Reply:
this is implemented